webbpsf/wfirst.py

```diff
@@ -323,5 +323,105 @@ def _validateConfig(self, **kwargs):
             pass
         super(WFI, self)._validateConfig(**kwargs)
 
+class CGI(WFIRSTInstrument):
+    """
+    WFIRST Coronagraph Instrument
+
+    """
+    def __init__(self, mode='CHARSPC', filter=None, pixelscale=None, fov_arcsec=None, apply_static_opd=False):
+        super(CGI, self).__init__("CGI", pixelscale=pixelscale)
+
+        self.pupil = os.path.join(self._WebbPSF_basepath, 'AFTA_CGI_C5_Pupil_onax_256px_flip.fits')
+        if apply_static_opd:
+            #self.pupilopd = os.path.join(self._WebbPSF_basepath, 'CGI', 'OPD', 'CGI_primary_OPD.fits')
+            self.pupilopd = os.path.join(self._WebbPSF_basepath, 'CGI', 'OPD', 'CGI_static_OPD.fits')
+        else:
+            self.pupilopd = None
+        self.mode_list = ['CHARSPC', 'DISKSPC']
+        self.image_mask_list = ['CHARSPC_F660', 'CHARSPC_F770', 'CHARSPC_F890', 'DISKSPC_F721']
+        self.pupil_mask_list = ['SPC30D88']
+        self.aberration_optic = None
+        self.options = {'force_coron':True}
+        # Allow the user to pre-emptively override the default instrument FoV and pixel scale
+        if fov_arcsec is not None:
+            self.fov_arcsec = fov_arcsec
+            self._override_fov = True
+        else:
+            self._override_fov = False
+        if pixelscale is not None:
+            self.pixelscale = pixelscale
+            self._override_pixelscale = True
+        else:
+            self._override_pixelscale = False
+        self.mode = mode
+        if filter is not None:
+            self.filter = filter
+        elif mode is 'CHARSPC':
+            self.filter = 'F770'
+        else:
+            self.filter = 'F721'
```

Comment:
Something else I was thinking about. I expect there are specific well-defined combinations of image mask, pupil mask, and filter which are allowed. Right? We could define a meta-mode attribute which acts to automatically set those appropriately and consistently. 

For instance with GPI we have defined e.g. obsmode="H_coron" mode means "H filter, H apodizer, H occulter, 080m12_04 Lyot".  Then in the GPI simulator class, if you set the obsmode attribute it automatically sets the filter, apodizer, occulter and lyot attributes. See https://github.com/geminiplanetimager/gpipsfs/blob/master/gpipsfs/main.py#L191 and https://github.com/geminiplanetimager/gpipsfs/blob/master/gpipsfs/main.py#L47

Would it make sense to lift that code for use in the CGI class?